### PR TITLE
[#2668] feat(all): Support updating default value of column

### PR DESCRIPTION
**Pull request by charliecheng630**
### What changes were proposed in this pull request?

Add UpdateColumnDefaultValue in TableChange for updating the default value of a field.

### Why are the changes needed?

Fix: #2668 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

New UTs.

### CODE REVIEW

**Comment by mchades** [comment]
You also need to add implementation in each catalog, and then add the integration tests and docs

File: catalogs/catalog-jdbc-postgresql/src/test/java/com/datastrato/gravitino/catalog/postgresql/integration/test/CatalogPostgreSqlIT.java [hunk 1]
```diff
@@ -747,6 +776,55 @@ void testColumnDefaultValue() {
...
+                table_comment,
+                createProperties());
+
+    catalog
+        .asTableCatalog()
+        .alterTable(
+            NameIdentifier.of(metalakeName, catalogName, schemaName, tableName),
+            TableChange.updateColumnDefaultValue(
+                new String[] {columns[0].name()},
+                toFunctionArg(Literals.of("1.234", Types.FloatType.get()))),
```
Comment (mchades) [inline catalogs/catalog-jdbc-postgresql/src/test/java/com/datastrato/gravitino/catalog/postgresql/integration/test/CatalogPostgreSqlIT.java:797]: It's unnecessary to use `toFunctionArg`

**Review by mchades — COMMENTED** [review]
Overall changes are right, you can proceed with other catalogs

File: catalogs/catalog-jdbc-postgresql/src/test/java/com/datastrato/gravitino/catalog/postgresql/integration/test/CatalogPostgreSqlIT.java (hunk 1 above)
Comment (charliecheng630) [reply catalogs/catalog-jdbc-postgresql/src/test/java/com/datastrato/gravitino/catalog/postgresql/integration/test/CatalogPostgreSqlIT.java:797]: The [Serde of ColumnDefaultValue](https://github.com/datastrato/gravitino/blob/main/common/src/main/java/com/datastrato/gravitino/json/JsonUtils.java#L1059) will cast the `LiteralImpl` to `FunctionArg`. If `toFunctionArg` is not used, an exception will be encountered.

Comment (mchades) [reply catalogs/catalog-jdbc-postgresql/src/test/java/com/datastrato/gravitino/catalog/postgresql/integration/test/CatalogPostgreSqlIT.java:797]: You should move `toFunctionArg` to [there](https://github.com/datastrato/gravitino/pull/2673/files#diff-49ca60933fb4234ec2833daacdf6ea31cf77579c5bc46e7aecdfd0adeacadadbR204)

**Comment by charliecheng630** [comment]
@mchades Please help review again, thank you.

File: common/src/main/java/com/datastrato/gravitino/dto/requests/TableUpdateRequest.java [hunk 2]
```diff
@@ -423,6 +427,60 @@ public TableChange tableChange() {
...
+     */
+    @Override
+    public void validate() throws IllegalArgumentException {
+      Preconditions.checkArgument(
+          fieldName != null
+              && fieldName.length > 0
+              && Arrays.stream(fieldName).allMatch(StringUtils::isNotBlank),
+          "\"fieldName\" field is required and cannot be empty");
+      Preconditions.checkArgument(
+          newDefaultValue != null, "\"newDefaultValue\" field is required and cannot be empty");
```
Comment (mchades) [inline common/src/main/java/com/datastrato/gravitino/dto/requests/TableUpdateRequest.java:474]: Is `DEFAULT_VALUE_NOT_SET` allowed?

Comment (charliecheng630) [reply common/src/main/java/com/datastrato/gravitino/dto/requests/TableUpdateRequest.java:474]: I add [a test for DEFAULT_VALUE_NOT_SET validation](https://github.com/datastrato/gravitino/pull/2673/commits/a43efcd9cf9895c18722fea9852283595622f770)  , it's allowed.

Comment (mchades) [reply common/src/main/java/com/datastrato/gravitino/dto/requests/TableUpdateRequest.java:474]: If doing so is meaningless, then I suggest throwing an exception during validation.

Comment (charliecheng630) [reply common/src/main/java/com/datastrato/gravitino/dto/requests/TableUpdateRequest.java]: Updated.

**Comment by mchades** [comment]
Please rebase main branch and resolve the conflicts when you have time, thx~

File: common/src/test/java/com/datastrato/gravitino/dto/requests/TestTableUpdatesRequest.java [hunk 3]
```diff
@@ -113,6 +132,19 @@ public void testTableUpdatesRequest() throws JsonProcessingException {
...
+    // test validate DEFAULT_VALUE_NOT_SET property value
+    TableUpdateRequest.UpdateTableColumnDefaultValueRequest updateTableColumnDefaultValueRequest =
+        new TableUpdateRequest.UpdateTableColumnDefaultValueRequest(
+            new String[] {"key"}, DEFAULT_VALUE_NOT_SET);
+    Assertions.assertThrows(
+        IllegalArgumentException.class, updateTableColumnDefaultValueRequest::validate);
+
+    updateTableColumnDefaultValueRequest =
+        new TableUpdateRequest.UpdateTableColumnDefaultValueRequest(
+            new String[] {"key"}, DEFAULT_VALUE_NOT_SET);
```
Comment (mchades) [inline common/src/test/java/com/datastrato/gravitino/dto/requests/TestTableUpdatesRequest.java:145]: duplicated with the above? should be null?

Comment (mchades) [reply common/src/test/java/com/datastrato/gravitino/dto/requests/TestTableUpdatesRequest.java:145]: BTW, also suggest to verify the exception message

**Review by mchades — APPROVED** [review]
LGTM, thanks for your contributions

**Pull request merged**